src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx

```diff
@@ -288,58 +305,37 @@ class DataPrivacyRulesPanelSelectorField extends React.Component<Props, State> {
     });
   };
 
-  handleKeyDown = debounce(
-    (event: React.KeyboardEvent<HTMLInputElement>) => {
-      event.persist();
-
-      const {keyCode} = event;
-      const {fieldValues, activeSuggestion, suggestions} = this.state;
-
-      if (keyCode === 8) {
-        const lastFieldValue = fieldValues[fieldValues.length - 1];
-        if (Array.isArray(lastFieldValue) && lastFieldValue[1].value.length === 1) {
-          this.setState({
-            fieldValues: [...fieldValues, lastFieldValue[0]],
-          });
-        }
-        return;
-      }
+  handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.persist();
 
-      if (keyCode === 13) {
-        this.handleClickSuggestionItem(suggestions[activeSuggestion])();
-        return;
-      }
+    const {keyCode} = event;
+    const {activeSuggestion, suggestions} = this.state;
 
-      if (keyCode === 38) {
-        if (activeSuggestion === 0) {
-          return;
-        }
-        this.setState({activeSuggestion: activeSuggestion - 1}, () => {
-          this.scrollToSuggestion();
-        });
-        return;
-      }
+    if (keyCode === 13) {
+      this.handleClickSuggestionItem(suggestions[activeSuggestion])();
+      return;
+    }
 
-      if (keyCode === 40) {
-        if (activeSuggestion === suggestions.length - 1) {
-          return;
-        }
-        this.setState({activeSuggestion: activeSuggestion + 1}, () => {
-          this.scrollToSuggestion();
-        });
+    if (keyCode === 38) {
+      if (activeSuggestion === 0) {
         return;
       }
+      this.setState({activeSuggestion: activeSuggestion - 1}, () => {
+        this.scrollToSuggestion();
+      });
+      return;
+    }
 
-      if (keyCode === 32) {
-        this.setState({
-          fieldValues: [...fieldValues, {value: ' ', type: 'string'}],
-        });
+    if (keyCode === 40) {
+      if (activeSuggestion === suggestions.length - 1) {
         return;
       }
-    },
-    DEFAULT_DEBOUNCE_DURATION,
```

Comment:
unfortunately I had to remove debounce for now, because  it was causing an issue when using keyDown and keyUp